captum/attr/_utils/gradient.py

```diff
@@ -293,23 +317,23 @@ def forward_hook(module, inp, out):
         # Identifies correct device ordering based on device ids.
         # key_list is a list of devices in appropriate ordering for concatenation.
         # If only one key exists (standard model), key list simply has one element.
-        key_list = _sort_key_list(list(saved_layer_outputs.keys()), device_ids)
-        all_outputs = _reduce_list(
-            [saved_layer_outputs[device_id] for device_id in key_list]
-        )
-        grad_inputs = tuple(saved_layer_outputs[device_id] for device_id in key_list)
+        key_list = _sort_key_list(list(saved_layer.keys()), device_ids)
+        all_outputs = _reduce_list([saved_layer[device_id] for device_id in key_list])
+        grad_inputs = tuple(saved_layer[device_id] for device_id in key_list)
         saved_grads = torch.autograd.grad(torch.unbind(output), grad_inputs)
         all_grads = torch.cat(saved_grads)
         if gradient_neuron_index is not None:
             inp_grads = _neuron_gradients(
-                inputs, saved_layer_outputs, key_list, gradient_neuron_index
+                inputs, saved_layer, key_list, gradient_neuron_index
             )
             return all_grads, all_outputs, inp_grads
         else:
             return all_grads, all_outputs
 
 
-def construct_neuron_grad_fn(layer, neuron_index, device_ids):
+def construct_neuron_grad_fn(
+    layer, neuron_index, device_ids=None, attribute_to_neuron_input=False
```

Comment:
@vivekmig , had to add `attribute_to_neuron_input ` attribute and also make `device_ids` optional.

Reply:
Ok, sounds good! Can you also pass the attribute_to_neuron_input flag in neuron_integrated_gradients? I think we'll also need to add the flag to neuron guided backprop / deconvnet, but I can do that in a separate PR if you prefer.